src/services/fileStorage/model.js

```diff
@@ -1,63 +1,61 @@
+ * @param isDirectory {Boolean} - is this a directory
  * @param name {String} - the name of the file, e.g. cloudy_pastel.jpeg
  * @param size {Number} - the size of the file in byte
  * @param type {String} - the type of the file, e.g. mime/image
- * @param flatFileName {String} - the name of the real file on the storage
+ * @param storageFileName {String} - the name of the real file on the storage
  * @param thumbnail {String} - the url of the file's thumbnail image
- * @param permissions [Permission] - given extra permission for this file (except the normal permissions)
- * @param lockId {ObjectId} - indicates whether a file is locked for editing or not (wopi-related)
  * @param shareToken {String} - hash for enabling sharing. if undefined than sharing is disabled
- * @param schoolId {ObjectId} - id of the school for file referencing
+ * @param parent {File} - parent directory
+ * @param owner {User|Course|Team} - owner Object of file
+ * @param permissions [Permission] - given permission for this file
+ * @param lockId {ObjectId} - indicates whether a file is locked for editing or not (wopi-related)
  */
 const fileSchema = new Schema({
-	key: {type: String, required: true, unique: true},
-	path: {type: String},
-	name: {type: String},
-	size: {type: Number},
-	type: {type: String},
-	flatFileName: {type: String},
-	thumbnail: {type: String},
-	permissions: [{
-		userId: {type: Schema.Types.ObjectId, ref: 'user'},
-		permissions: [{type: String, enum: permissionTypes}]
-	}],
-	lockId: {type: Schema.Types.ObjectId},
-	shareToken: {type: String},
-	studentCanEdit: {type: Boolean, default: false} ,
-	schoolId: [{type: Schema.Types.ObjectId, ref: 'school'}],
-	createdAt: {type: Date, 'default': Date.now},
-	updatedAt: {type: Date, 'default': Date.now}
-});
-
-/**
- * handles meta-data for a directory
- * @param key {String} - the key/path to the directory, e.g. users/0000d231816abba584714c9e/folder
- * Information: key could be redundant because it is just path + name, but it is easier to handle different server calls
- * @param path {String} - the path in which the file exists users/0000d231816abba584714c9e/
- * @param name {String} - the name of the file, e.g. folder
- */
-const directorySchema = new Schema({
-	key: {type: String, required: true, unique: true},
-	path: {type: String},
-	name: {type: String},
-	createdAt: {type: Date, 'default': Date.now},
-	updatedAt: {type: Date, 'default': Date.now}
+	isDirectory: { type: Boolean, default: false },
+	name: { type: String },
+	size: { type: Number },
+	type: { type: String },
+	storageFileName: { type: String },
+	thumbnail: { type: String },
+	shareToken: { type: String },
+	parent: { type: Schema.Types.ObjectId, ref: 'file' },
+	owner: {
+		type: Schema.Types.ObjectId,
+		required: true,
+		refPath: 'refOwnerModel'
+	},
+	refOwnerModel: {
+		type: String,
+		required: true,
+		enum: ['user', 'course', 'teams']
```

Comment:
For teams we must add a additional hook / handling that ask the team services with get to take the additional permissions for this team. See the model key filePermission. It has the same permissionSchema. By team create it initial start with one array element for each role and default values.